### PR TITLE
Haskell fixes

### PR DESCRIPTION
**Pull request by chrisbarrett**
Should fix indentation-related issues, but the bug reports are kinda all over the place so we'll have to merge and see.

### CODE REVIEW

File: contrib/lang/haskell/packages.el [hunk 1]
```diff
@@ -178,7 +159,30 @@
...
         '(define-key haskell-cabal-mode-map
-           [?\C-c ?\C-z] 'haskell-interactive-switch)))))
+           [?\C-c ?\C-z] 'haskell-interactive-switch))))
+
+  (use-package haskell-indentation
+    :defer t
+    :config
+    (progn
+      ;; Show indentation guides in insert or emacs state only.
+      (defun spacemacs//haskell-identation-show-guides ()
```
Comment (ajknoll) [inline contrib/lang/haskell/packages.el]: Are these named `identation` rather than `indentation` deliberately?

Comment (chrisbarrett) [inline contrib/lang/haskell/packages.el]: That looks like a typo. I'll push a fix.

**Comment by srid** [comment]
@chrisbarrett Could you please share your `.spacmacs`? I'm testing this locally, and indentation still doesn't work for me.

**Comment by srid** [comment]
Specifically, after applying your patch:

* the value of my `haskell-mode-hook`'s value is `(structured-haskell-mode ghc-init)`
* yet hitting C-j continues to complain about indentation mode not being enabled

screenshot below:

screen shot 2015-04-17 at 12 27 30 am

**Comment by chrisbarrett** [comment]
My spacemacs config is unlikely to be helpful because my haskell config has diverged a lot from spacemacs. It's on github anyway. https://github.com/chrisbarrett/.emacs.d/tree/master/private

**Comment by chrisbarrett** [comment]
Does `(add-hook 'haskell-mode-hook 'turn-on-haskell-indentation)` help in your case?

**Comment by chrisbarrett** [comment]
Have you got `(setq haskell-enable-shm-support t)` in your config?

**Comment by chrisbarrett** [comment]
What command is being run when you hit `C-j`? Is it `newline-and-indent`?

**Comment by srid** [comment]
C-j runs `electric-newline-and-maybe-indent`.

My spacemacs config looks like:

```
(setq-default dotspacemacs-configuration-layers
              '(markdown
                git
                javascript
                (haskell :variables haskell-enable-shm-support t)))
```

Manually doing that add-hook think had no effect.

**Comment by chrisbarrett** [comment]
Can we move this conversation to gitter? :)

**Comment by chrisbarrett** [comment]
The above was a configuration problem, and is resolved.

**Comment by syl20bnr** [comment]
Thank you ! :+1:
I will cherry-picked it in 0.101 if it is OK.
I changed the `use-package` for an `eval-after-load`, while `use-package` does the job and is perfectly valid I found it weird since `haskell-indentation` is part of the `haskell-mode` package.

Cherry-picked into `develop` branch, you can safely delete your branch.

**Pull request closed without merging**